contracts/ColonyTask.sol

```diff
@@ -266,13 +260,12 @@ contract ColonyTask is ColonyStorage, DSMath {
     return taskWorkRatings[_id].secret[_role];
   }
 
-  // TODO: Restrict function visibility to whoever submits the approved Transaction from Client
   // Note task assignment is agreed off-chain
   function setTaskRoleUser(uint256 _id, uint8 _role, address _user) public
```

Comment:
Ok so we are securing this down to the manager only. That means that they can change the assigned worker at any point which is not right. Have you discussed this scenario with Pat is this what they want?

Reply:
As i said in PR description, Pat decided to delegate task level permissions to us, developers. You can also see in the description who can call which function, so if you have different opinion, we can change it.

Reply:
Can you also see every other task related function. You can see just the description first, and the code later

Reply:
You mean in the PR description? I can't comment against the description to extracting it here for feedback (if that's what you meant):

>setTaskSkill - Since everybody will receive the reputation in that skill, this should be probably agreed by everyone? Right now it is agreed between manager and worker.

Not everyone receives rep in the global skill, just the worker. It's probably best to leave it as-is to be agreed between the manager and worker.

>setTaskDomain - Since worker is going to receive reputation in that domain, manager and worker should agree.

Every role received domain rep actually but since the manager determines the domain with the task creation I don't see a point in restricting that down to other roles if they decided to change it.

Reply:
>setTaskManagerPayout - Can only be called by manager. Should we allow manager to set his payout?

Yes, I think they should be able to determine their own payout.

Reply:
> setTaskRoleUser
Ideally we would get mutual approval between the manager and whichever role is being assigned here. Not sure how possible/expensive is this in code as `executeTaskChange` relies on the reviewer roles being present to execute. In cases where one is missing, the other one executes alone. Can you find out?

Reply:
Ok, i switched `setTaskDomain` and `setTaskSkill`, sorry: So,
- `setTaskSkill` - manager and worker.
- `setTaskDomain` - manager.

Reply:
Can you have a separate commit testing how the implementation of `setTaskRoleUser` would work? Then we can decide whether to include it. The main problem with this is the `executeTaskChange` implementation which won't cover this scenario in its current shape, i.e. `setTaskWorker` would require manager and worker to agree but worker is not assigned here as the function itself does that.

Reply:
ok for `setTaskDomain` and `setTaskSkill`